src/gluonts/dataset/__init__.py

```diff
@@ -14,5 +14,25 @@
 # !!! DO NOT MODIFY !!! (pkgutil-style namespace package)
 
 from pkgutil import extend_path
+from .common import ListDataset
+from .field_names import FieldName
+from .repository.datasets import get_dataset, dataset_recipes, load_datasets
```

Comment:
I think I'm against making these available directly.

In the future, I would like to see `repository` be moved out of `dataset`.